frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts

```diff
@@ -66,28 +78,119 @@ describe('KubeVirt CDI Upload', () => {
       await uploadForm.upload(pvc);
       await browser.wait(until.presenceOf(errorMessage));
       expect(errorMessage.getText()).toContain('not supported');
-    },
-    CDI_UPLOAD_TIMEOUT_SECS,
-  );
+    });
 
-  it(
-    'ID(CNV-4890) Upload image for golden OS',
-    async () => {
-      const pvc: PVCData = {
-        image: desImage,
-        os: OperatingSystem.RHEL7,
-        pvcSize: '1',
-        pvcSizeUnits: 'Gi',
-        storageClass: STORAGE_CLASS,
-      };
+    it(
+      'ID(CNV-5038) Upload data to golden OS template again after delete',
+      async () => {
+        await withResource(leakedResources, rhel7PVC.getDVResource(), async () => {
+          await rhel7PVC.create();
+        });
+        // it can be associated again.
+        await withResource(leakedResources, rhel7PVC.getDVResource(), async () => {
+          await rhel7PVC.create();
+        });
+      },
+      2 * CDI_UPLOAD_TIMEOUT_SECS,
+    );
 
-      await uploadForm.upload(pvc);
-      // It has to click the upload button again to trigger uploading based on actual test results.
-      await click(saveChangesBtn);
-      await browser.wait(until.textToBePresentInElement(cdiUploadView.uploadProgress, '100%'));
-      await click(cdiUploadView.viewStatusID);
-      await browser.wait(until.textToBePresentInElement(pvcStatus, PVC_STATUS.BOUND));
-    },
-    CDI_UPLOAD_TIMEOUT_SECS,
-  );
+    it('ID(CNV-5176) It shows an error when uploading data to golden OS again', async () => {
+      await withResource(leakedResources, rhel7PVC.getDVResource(), async () => {
+        await rhel7PVC.create();
+        await uploadForm.openForm();
+        await uploadForm.selectGoldenOS(rhel7PVC.os);
+        await browser.wait(until.presenceOf(errorMessage));
+        expect(errorMessage.getText()).toContain('Operating system source already defined');
+      });
+    });
+  });
+
+  describe('KubeVirt GOLDEN OS Creation', () => {
+    const fedoraPVC = new PVC(FEDORA_PVC);
+    const win10PVC = new PVC(WIN10_PVC);
+
+    beforeAll(async () => {
+      imagePull(FEDORA_IMAGE, fedoraPVC.image);
+      imagePull(WIN10_IMAGE, win10PVC.image);
+    });
+
+    afterAll(async () => {
+      execSync(`rm ${FEDORA_PVC.image} ${WIN10_PVC.image}`);
+    });
+
+    it(
+      'ID(CNV-5042) Create multiple VMs from the same golden os template',
+      async () => {
+        const vm1 = new VMBuilder(getBasicVMBuilder())
+          .setOS(rhel7PVC.os)
+          .generateNameForPrefix('auto-clone-vm1')
+          .setStartOnCreation(true)
+          .build();
+
+        const vm2 = new VMBuilder(getBasicVMBuilder())
+          .setOS(rhel7PVC.os)
+          .generateNameForPrefix('auto-clone-vm2')
+          .setStartOnCreation(true)
+          .build();
+
+        await withResource(leakedResources, rhel7PVC.getDVResource(), async () => {
```

Comment:
I am still not seeing the disposal of leakedResources

Reply:
Done, added it into afterAll.